### PR TITLE
backend/sdoc_source_code: JUnit XML test report reader

### PR DESCRIPTION
**Pull request by stanislaw**
- [x] Backward slashes on Windows.
- [x] Status for Skipped
- [x] Status for Failed

Postponed:
- CTest XML vs Google Test XML.
- Edge case: An itest is in the root of a test folder.
- StrictDoc's own itests without and with HTML2PDF. How to merge the test reports.

### CODE REVIEW

File: strictdoc/backend/sdoc_source_code/reader_python.py [hunk 1]
```diff
@@ -203,6 +205,14 @@ def read(
                 parent_function.child_functions.append(new_function)
                 functions_stack.append(new_function)
                 traceability_info.functions.append(new_function)
+
+                traceability_info.ng_map_names_to_definition_functions[
```
Comment (stanislaw) [inline strictdoc/backend/sdoc_source_code/reader_python.py:209]: Most probably remove

File: strictdoc/core/file_traceability_index.py [hunk 2]
```diff
@@ -173,6 +173,86 @@ def get_coverage_info(
...
+                                    file_path_posix
+                                )
+
+                    node.set_field_value(
+                        field_name="TEST_PATH",
+                        form_field_index=0,
+                        value=resolved_path_to_function_file,
+                    )
+
+                    self.create_requirement(node, overwrite=True)
```
Comment (stanislaw) [inline strictdoc/core/file_traceability_index.py:230]: Double-check if this is needed.

File: strictdoc/export/html/templates/screens/project_index/project_tree_folder.jinja [hunk 3]
```diff
@@ -19,7 +19,7 @@
     {% endfor %}
 
     {% for file_ in folder.files %}
-      {% if file_.has_extension(".sdoc") %}
+      {% if file_.has_extension(".sdoc") or file_.has_extension(".junit.xml") %}
```
Comment (stanislaw) [inline strictdoc/export/html/templates/screens/project_index/project_tree_folder.jinja:22]: Double-check that on the single document screen, the reports are also displayed.

File: strictdoc/helpers/google_test.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+def convert_function_name_to_gtest_macro(input_str: str) -> str:
+    assert isinstance(input_str, str)
```
Comment (stanislaw) [inline strictdoc/helpers/google_test.py:2]: Add an example of a conversion to """.

File: tests/integration/features/test_reports/junit_xml/pytest/03_skipped_test/strictdoc.toml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[project]
+
+features = [
+  "REQUIREMENT_TO_SOURCE_TRACEABILITY",
+  "SOURCE_FILE_LANGUAGE_PARSERS",
+]
+
+exclude_source_paths = [
```
Comment (stanislaw) [inline tests/integration/features/test_reports/junit_xml/pytest/03_skipped_test/strictdoc.toml:8]: Double-check if this is needed.

File: tests/integration/lit.cfg [hunk 6]
```diff
@@ -5,7 +5,7 @@ import sys
 import lit.formats
 
 
-config.name = "StrictDoc integration tests"
+config.name = "StrictDoc Integration Tests"
```
Comment (stanislaw) [inline tests/integration/lit.cfg:8]: Remove this hunk.
